landing-page/content/common/how-to-release.md

```diff
@@ -329,16 +372,37 @@ git rebase main
 git push apache latest
 ```
 
-#### Set latest version in iceberg-docs repo
+## Finalize release
+
+### Send announcement email
+
+After every step is completed, send an announcement email:
+
+```text
+[ANNOUNCE] Apache Iceberg release <VERSION>
+```
+```text
+I'm pleased to announce the release of Apache Iceberg <VERSION>!
+
+Apache Iceberg is an open table format for huge analytic datasets. Iceberg
+delivers high query performance for tables with tens of petabytes of data,
+along with atomic commits, concurrent writes, and SQL-compatible table
+evolution.
+
+This release can be downloaded from: https://www.apache.org/dyn/closer.cgi/iceberg/<TARBALL NAME WITHOUT .tar.gz>/<TARBALL NAME>
+
+Java artifacts are available from Maven Central.
+
+Thanks to everyone for contributing!
+```
 
-The last step is to update the `main` branch in `iceberg-docs` to set the latest version.
-A PR needs to be published in the `iceberg-docs` repository with the following changes:
-1. Update variable `latestVersions.iceberg` to the new release version in `landing-page/config.toml`
-2. Update variable `latestVersions.iceberg` to the new release version and 
-`versions.nessie` to the version of `org.projectnessie.nessie:*` from [versions.props](https://github.com/apache/iceberg/blob/master/versions.props) in `docs/config.toml`
-3. Mark the current latest release notes to past releases under `landing-page/content/common/release-notes.md`
-4. Add release notes for the new release version in `landing-page/content/common/release-notes.md`
+### Update Iceberg codebase
 
+After the release artifacts are available in Maven, 
+there are a few places in the codebase that reference the latest Iceberg release version number and need to be updated:
+1. Update `options` list for `Apache Iceberg version` in https://github.com/apache/iceberg/blob/master/.github/ISSUE_TEMPLATE/iceberg_bug_report.yml
+2. Update `ICEBERG_VERSION` in https://github.com/apache/iceberg/blob/master/python/dev/Dockerfile
+3. Update `oldVersion` in `revapi` in https://github.com/apache/iceberg/blob/master/build.gradle
```

Comment:
Looks like we should make a note to not do this for patch release:  https://github.com/apache/iceberg/pull/8150#discussion_r1274694642